Framework/Algorithms/src/GetEi2.cpp

```diff
@@ -532,6 +532,16 @@ double GetEi2::calculatePeakWidthAtHalfHeight(const API::MatrixWorkspace_sptr &d
                       << "half-height point will not be as accurate.\n";
       ip1--;
     }
+    // similarly, noise may cause two points around the
+    while (peak_y[ip1] == peak_y[ip2]) { // TODO: shold be smoothing with runnign average here
```

Comment:
Remove the TODO and either:
- Improve the implementation as part of this PR
- Add a new github issue detailing the idea you had

Reply:
This are fair comments, but unfortunately I do not have time to make such substantial changes. I have user request to fulfill and very short time to address. 

Would be nice to have unit test, bit the function is very specific

Reply:
Probably one needs to take the data from the dataset and hardcode them to the unit test